src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -93,7 +93,7 @@ public void Initialize()
 
         public void Invalidate()
         {
-            SetDescriptionIfChanged(_baseDescription);
+            SetDescriptionIfChanged(_baseDescription.With(lastUpdateTimestamp: DateTime.UtcNow));
```

Comment:
Strictly speaking this change isn't needed anymore now that we're not checking for timestamps when we update the topology, but I think updating the timestamp is a bit more correct and can help with debugging.

Reply:
We agreed that we want to update the timestamp every time a new ServerDescription is created so that we can detect in exception error messages whether the descriptions are stale.

Reply:
FYI: ServerDescriptionIfChanged is probably obsolete because now that we use timestamps the description is always different even if the timestamp is the only thing that changed.

Reply:
Agreed: I've removed the method and replaced invocations of it with SetDescription